packages/react-components/react-virtualizer/src/hooks/useScrollRef.ts

```diff
@@ -0,0 +1,43 @@
+import * as React from 'react';
+import { debounce } from '../utilities/debounce';
+import { canUseDOM } from '@fluentui/react-utilities';
+import { ResizeCallbackWithRef } from './hooks.types';
+
+/**
+ * useScrollRef simplifies resize observer connection and ensures debounce/cleanup
+ */
+export const useScrollRef = (resizeCallback: ResizeCallbackWithRef) => {
+  const container = React.useRef<HTMLElement | null>(null);
+  // the handler for resize observer
+  const handleResize = debounce((entries: ResizeObserverEntry[], observer: ResizeObserver) => {
+    resizeCallback(entries, observer, container);
+  });
+
+  // Keep the reference of ResizeObserver in the state, as it should live through renders
+  const [resizeObserver] = React.useState(() => (canUseDOM() ? new ResizeObserver(handleResize) : undefined));
```

Comment:
you can ignore this if it's intentional.

Currently if the user changes the `handleResize` function passed to this hook, the resize observer will never update.

https://codesandbox.io/s/sad-matsumoto-6rcpul?file=/src/App.js



```